iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs

```diff
@@ -52,142 +52,113 @@ public static Exception GetExceptionFromOutcome(Outcome outcome)
             return retException;
         }
 
-        public static Exception ToIotHubClientContract(AmqpException amqpException)
+        public static IotHubClientException ToIotHubClientContract(AmqpException amqpException)
         {
             Error error = amqpException.Error;
             AmqpSymbol amqpSymbol = error.Condition;
             string message = error.ToString();
 
+            IotHubClientException retException;
+
             // Generic AMQP error
             if (Equals(AmqpErrorCode.InternalError, amqpSymbol))
             {
-                return new IotHubClientException(message, IotHubClientErrorCode.NetworkErrors, amqpException);
+                retException = new(message, IotHubClientErrorCode.NetworkErrors, amqpException);
             }
             else if (Equals(AmqpErrorCode.NotFound, amqpSymbol))
             {
-                return new IotHubClientException(message, IotHubClientErrorCode.DeviceNotFound, amqpException);
+                retException = new(message, IotHubClientErrorCode.DeviceNotFound, amqpException);
             }
             else if (Equals(AmqpErrorCode.UnauthorizedAccess, amqpSymbol))
             {
-                return new IotHubClientException(message, IotHubClientErrorCode.Unauthorized, amqpException);
-            }
-            else if (Equals(AmqpErrorCode.DecodeError, amqpSymbol))
-            {
-                return new IotHubClientException(message, amqpException);
+                retException = new(message, IotHubClientErrorCode.Unauthorized, amqpException);
             }
             else if (Equals(AmqpErrorCode.ResourceLimitExceeded, amqpSymbol))
             {
-                return new IotHubClientException(message, IotHubClientErrorCode.DeviceMaximumQueueDepthExceeded, amqpException);
-            }
-            else if (Equals(AmqpErrorCode.NotAllowed, amqpSymbol))
-            {
-                return new InvalidOperationException(message, amqpException);
-            }
-            else if (Equals(AmqpErrorCode.InvalidField, amqpSymbol))
-            {
-                return new InvalidOperationException(message, amqpException);
-            }
-            else if (Equals(AmqpErrorCode.NotImplemented, amqpSymbol))
-            {
-                return new NotSupportedException(message, amqpException);
```

Comment:
these will be addressed in the "else" block below and returned as `IotHubClientException`.